pkg/optimizer/flatzinc_model.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package optimizer
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"strings"
+)
+
+type FlatZincParam struct {
```

Comment:
General comment - please provide documentation for all structures and functions, and please add a link to FlatZinc

Reply:
done